### DISCUSSION TITLE
I can't authenticate a user in a Next.JS API route with setSession due to RLS

### DISCUSSION THREAD

**Discussion by react-ghost** [supabase/supabase#10056] [Questions]
I am trying to perform some server side work that requires me to update the user on the server side in a **Next.JS** api route. The requests are coming from a **React Native** client.

The general flow of what I'm trying to do looks like this:
```
// access and refresh tokens being sent with request
{ error } = await supabase.auth.setSession({ access_token, refresh_token })

// do some stuff and get some data to update users table here:

const { data, error } = await supabase.from('users').update({ fieldToUpdate }).match({ id: userID });
```

error prints:

`{ message: '' }`

When I remove the RLS, it works fine. My RLS rule is a template that looks like this:

```
CREATE POLICY "policy_name"
ON public.users
FOR UPDATE USING (
  auth.uid() = user_id
);
```

I had seen a few other discussion on this, and it appears that this was supposed to be a away that I was able to authenticate for a short period to perform this work on the server side. I do not want to bypass RLS by creating a client with the service key because I have the opportunity to provide the tokens from the client. it seems I should be able to do this this way.

Is there something I'm missing?

**Comment by GaryAustin1** [comment]
You will need to use a client with the service role key (server side) to bypass RLS.  You can change the authorization header to the service role to do that.

**Reply by react-ghost** [reply]
I'm sorry I may not be understanding. If I'm making a request to my Next API route from a React Native client, is there no way to verify the authentication of the user making the request from the RN Client? Do i have to use a service key and bypass RLS entirely, or is there anyway I can authenticate the request being made in the Next API route? 

Because If I use the service key that essentially leaves the next API route unprotected from an unauthorized supabase request from any client, so if someone got ahold of a user ID, they could manipulate data through my API endpoint if it's using the service key.

I thought I could set a supabase auth session using supabase.auth.setSession and pass the access_token and refresh_token through the request such that supabase would set the session, and thus allow the table operations assuming the user making the request is able to perform those operations according to the RLS rules I've set.

**Reply by GaryAustin1** [reply]
Apologies,  I misunderstood your question.  I thought you were trying to bypass RLS temporarily. 

This seems to support the method you are using.  
https://supabase.com/docs/guides/auth/server-side-rendering
